### PR TITLE
Support Swift Package Manager

### PR DESCRIPTION
**Pull request by Nixon506E**

### CODE REVIEW

**Comment by hyouuu** [comment]
@zhongwuzw @SD10 @nathantannar4  could you take a look at PRs? Or maybe add a few members to keep this project alive?

**Comment by zhongwuzw** [comment]
@hyouuu Hey, sorry we don't have much time on it, would u want to collaborate to help us? or any members you can recommend?

**Comment by hyouuu** [comment]
Yes I would love to at least do PR reviews so the community can help move it forward :)

**Comment by hyouuu** [comment]
Didn't pay close attention to contributions but you could look into PR submission etc to see who would be the best candidates

**Comment by zhongwuzw** [comment]
> Yes I would love to at least do PR reviews so the community can help move it forward :)

I added you and @moldovaniosif write permission of master, please help us to move on. :)

**Comment by hyouuu** [comment]
Nice thanks!

**Comment by hyouuu** [comment]
@zhongwuzw would you be able to grant access to circle so I can re-run test for https://github.com/MessageKit/MessageKit/pull/1189

File: Package.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+let package = Package(
+   name: "MessageKit",
+   platforms: [.iOS(.v9)],
+   products: [
+       .library(name: "MessageKit",
+                targets: ["MessageKit"])
+   ],
+   dependencies: [
+        .package(url: "https://github.com/Nixon506E/InputBarAccessoryView.git", from: "4.2.1")
```
Comment (hyouuu) [inline Package.swift:12]: Saw your branch is 2 commits ahead - assuming you'll keep it updated :)

File: Package.resolved [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        "package": "InputBarAccessoryView",
+        "repositoryURL": "https://github.com/Nixon506E/InputBarAccessoryView.git",
+        "state": {
+          "branch": null,
+          "revision": "3d2ffabd19b83ae3f8d609d1d49f9a7e72883bbd",
+          "version": "4.3.3"
+        }
+      },
+      {
+        "package": "RxSwift",
```
Comment (hyouuu) [inline Package.resolved:14]: Do we actually use RxSwift in MessageKit?

Comment (Kaspik) [reply Package.resolved:14]: Soo, that's something we should talk about and take action asap - there is RxSwift dependency for InputBarAccessory but I believe it's wrong.

File: Package.swift (hunk 1 above)
Comment (Kaspik) [reply Package.swift:12]: Please open PR to the original repo - this can't be merged I believe.

File: Package.resolved (hunk 2 above)
Comment (Kaspik) [reply Package.resolved:14]: RxSwift should be taken away and aded as an optional dependency / extension.

**Review by Kaspik — CHANGES_REQUESTED** [review]

**Comment by Kaspik** [comment]
Also please point to the `development` branch, not master.

**Comment by xcodefriend** [comment]
@austinwright , Can we get this PR merged? Really need this feature.

**Comment by Kaspik** [comment]
We can't yet, it's not even up-to-date, as well as we should split this and RxSwift extension so people with SPM don't hve Rx included.

**Comment by xcodefriend** [comment]
@Kaspik, Can I help to speed up the process?

**Comment by Kaspik** [comment]
@xcodefriend It would be nice to find a way how to have RxMessageKit split out of main project with main project as a dependency, so we can have separate Package file for that and separate for main project without RxExtensions completely. What do you think?
(SPM doesn't have support for "subspec" kind of things so Rx would be always included even for people without that - that's why we need to make these changes)

**Comment by Kaspik** [comment]
Closing in favor of https://github.com/MessageKit/MessageKit/pull/1341.

**Pull request closed without merging**